gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -118,12 +118,12 @@ def self.signature_for_method(method)
   # Only public because it needs to get called below inside the replace_method blocks below.
   def self._on_method_added(hook_mod, method_name, is_singleton_method: false)
     current_declaration = T::Private::DeclState.current.active_declaration
-    return if !current_declaration
-    T::Private::DeclState.current.reset!
-
     mod = is_singleton_method ? hook_mod.singleton_class : hook_mod
     original_method = mod.instance_method(method_name)
 
+    return if !current_declaration
+    T::Private::DeclState.current.reset!
+
```

Comment:
I imagine the way this was written before was for perf reasons, so that we didn't have to box up the `instance_method` if we knew we were going to early exit. Why did this move down?

Reply:
Because later, we want to look at those variables to do the final checks, which (as I understand) need to be done for every method def, even if the method def doesn't have a sig to go with it.

I tried to do the pre-work-y changes that don't change any semantics before actually implementing the checks. Because of that, the first few commits in this PR are basically small random changes that seem irrelevant. This is an example of that.

But I committed them first so that the actual commit that adds the new logic is purely additive (no red in the diff) so one can focus on what was actually added. Sorry it's a bit confusing.

Should I not bother trying to separate out these tiny changes?

Reply:
If i need to look at a future commit to understand a given commit, it's more confusing for me than if they were in one big commit.

Reply:
Ok, I'll avoid that going forward